openapi.yaml

```diff
@@ -4,93 +4,285 @@ info:
+        See: #/components/schemas/DisplayName
+      type: string
+
+    PhaseReference:
+      description: To reference a phase, its generation and order is needed.
+      type: object
+      properties:
+        displayName:
+          $ref: '#/components/schemas/DisplayName'
+        order:
+          $ref: '#/components/schemas/Incremental'
+        generation:
+          $ref: '#/components/schemas/Incremental'
+      required:
+        - order
+        - generation
+
+    PhaseList:
+      description: Phase resources are always handled as a list.
+      type: object
+      properties:
+        generation:
+          $ref: '#/components/schemas/Incremental'
+        phases:
+          type: array
+          items:
+            $ref: '#/components/schemas/Phase'
+      required:
+        - generation
+        - phases
+
     IncidentUpdate:
+      description: |
+        An update is a sub resource to an incident.
+        It's identified by the incident ID and its own order.
+        Updates happen in a given order.
       type: object
-      required: [text, createdAt]
       properties:
-        text:
-          type: string
+        order:
+          $ref: '#/components/schemas/Incremental'
+        displayName:
+          $ref: '#/components/schemas/DisplayName'
+        description:
+          $ref: '#/components/schemas/Description'
         createdAt:
-          type: string
-          format: date-time
+          $ref: '#/components/schemas/Date'
+      required:
+        - order
+        - createdAt
+
     Labels:
+      description: Labels are free text key value pairs for components.
       type: object
       additionalProperties:
         type: string
+
     Component:
       type: object
-      required:
-        - id
-        - displayName
-        - labels
-        - affectedBy
       properties:
         id:
-          type: string
+          $ref: '#/components/schemas/Id'
         displayName:
-          type: string
+          $ref: '#/components/schemas/DisplayName'
         labels:
           $ref: '#/components/schemas/Labels'
         affectedBy:
-          type: array
-          items:
-            $ref: '#/components/schemas/Id'
-    Incident:
-      type: object
+          $ref: '#/components/schemas/ImpactList'
       required:
         - id
-        - title
-        - description
-        - updates
-        - affects
-        - impactType
-        - phase
+
+    Incident:
+      type: object
       properties:
         id:
-          type: string
-        title:
-          type: string
+          $ref: '#/components/schemas/Id'
+        displayName:
+          $ref: '#/components/schemas/DisplayName'
         description:
-          type: string
+          $ref: '#/components/schemas/Description'
         updates:
-          type: array
-          items:
-            $ref: '#/components/schemas/IncidentUpdate'
+          $ref: '#/components/schemas/Incremental'
```

Comment:
Rather `array` of `Incrementals`? Also maybe marking as `readOnly`, if possible?